### PR TITLE
Register form validations

### PR DESCRIPTION
**Pull request by dbrosio3**
## Links:

* [Trello Card](https://trello.com/c/5QOv8lHL/2-2-as-a-user-i-should-be-able-to-sign-up-to-create-an-account)
* [Mockups](https://www.figma.com/file/79uiALRVOHTMEHyqShdHnA/Black-Market?node-id=1%3A3)

## What & Why:

Added registration form schema 
Added form controller 
Added error messages

image
image

## Risks, Mitigation & Rollback:

- [x] Safe to Revert *check this box if this PR can be reverted without incident*

### CODE REVIEW

File: src/features/auth/utils/constants.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export const FULL_NAME_MIN_LENGTH = 2;
+export const FULL_NAME_MAX_LENGTH = 50;
+
+export const USERNAME_MIN_LENGTH = 2;
+export const USERNAME_MAX_LENGTH = 50;
+
+export const PASSWORD_REGEX = new RegExp('^(?=.*[A-Za-z])(?=.*\\d)[A-Za-z\\d]{8,}$');
```
Comment (dbrosio3) [inline src/features/auth/utils/constants.js:7]: I just improvised some placeholder constraints but I'm not sure if there's any definition about this.

File: src/components/Form/TextField.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+type TextFieldProps = {
+  name: string;
+  label: string;
+  placeholder: string;
+  icon?: ReactNode;
+};
+
+export const TextField: React.FC<TextFieldProps> = ({ label, ...props }) => {
+  console.table(props);
```
Comment (flomosq) [inline src/components/Form/TextField.tsx]: Unless you need it right now I'd delete it
```suggestion
  console.table(props);
```

File: src/components/Form/TextField.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  const [field, meta] = useField(props);
+
+  // simple condition to add spacing when the help text is too long
+  const errorMessageHeight = meta.error && meta.error.length > 43 ? 8 : 2;
```
Comment (flomosq) [inline src/components/Form/TextField.tsx]: If you might use this somewhere else I'd create a custom hook

**Review by flomosq — APPROVED** [review]
🚀

File: src/components/Form/TextField.tsx (hunk 2 above)
Comment (dbrosio3) [reply src/components/Form/TextField.tsx]: yep, forgot to remove it. Thanks!

File: src/components/Form/TextField.tsx (hunk 3 above)
Comment (dbrosio3) [reply src/components/Form/TextField.tsx]: That's a neat idea

File: src/features/auth/utils/registerFormSchema.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  password: z.string().regex(PASSWORD_REGEX),
+});
+
+/* schema adapted to formik */
+export const registerFormikSchema = toFormikValidationSchema(registerFormSchema);
+
+/* localized error messages */
+const tp = translationWithPrefix('error.validations');
+
+const customErrorMap: z.ZodErrorMap = (issue, ctx) => {
```
Comment (dariokozicki) [inline src/features/auth/utils/registerFormSchema.ts]: I feel like you can put this logic in the registerFormSchema using refinement and custom errors:
`z.string().min(5, { message: "Must be 5 or more characters long" });`

`const myString = z.string().refine((val) => val.length <= 255, {
  message: "String can't be more than 255 characters",
});`

**Review by dariokozicki — APPROVED** [review]
Good work!

File: src/features/auth/utils/registerFormSchema.ts (hunk 4 above)
Comment (dbrosio3) [reply src/features/auth/utils/registerFormSchema.ts]: That's a great suggestion. I attempted it like this at first but was getting errors from z and typescript. 

I finally figured it out :)

**Pull request merged**